xray/commands/curation/analyse.go

```diff
@@ -0,0 +1,170 @@
+package curation
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	utils2 "github.com/jfrog/jfrog-cli-core/v2/xray/commands/utils"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/xray/services/utils"
+	"net/http"
+	"strings"
+)
+
+type treeAnalyzer struct {
+	artiManager       artifactory.ArtifactoryServicesManager
+	artAuth           auth.ServiceDetails
+	httpClientDetails httputils.HttpClientDetails
+	url               string
+	repo              string
+	tech              coreutils.Technology
+}
+
+const (
+	BlockMessageKey  = "jfrog packages curation"
+	NotBeingFoundKey = "not being found"
+)
+
+func (nc *treeAnalyzer) recursiveNodeCuration(graph *utils.GraphNode, respStatus *[]PackageStatus, parent, parentVersion string, isRoot bool) error {
+	if parent == "" && !isRoot {
+		_, parent, parentVersion = getUrlNameAndVersionByTech(nc.tech, graph.Id, nc.url, nc.repo)
+	}
+	for _, node := range graph.Nodes {
+		packageUrl, name, version := getUrlNameAndVersionByTech(nc.tech, node.Id, nc.url, nc.repo)
+		resp, _, err := nc.artiManager.Client().SendHead(packageUrl, &nc.httpClientDetails)
+		if err != nil || resp.StatusCode >= 400 {
+			if resp == nil {
+				return err
```

Comment:
We can potentially return a nil error here. To avoid this, we should wrap this potential error with anbother error, wuth a message tha includes content. This error should of course be wrappper with `CheckError`.